__tests__/maintenance.ts

```diff
@@ -21,27 +21,18 @@
  */
 import { DateTime } from 'luxon'
 
-import { handleRequest as f } from '../src'
-import { contentTypeIsHtml, containsText } from './utils'
-
-let fetchMock: jest.Mock
-
-beforeEach(() => {
-  fetchMock = jest.fn(() => {
-    return true
-  })
-  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-  // @ts-ignore
-  window['fetch'] = fetchMock
-})
+import { handleRequest } from '../src'
+import { createApiResponse } from './frontend-proxy'
+import { contentTypeIsHtml, containsText, withMockedFetch } from './utils'
 
 describe('Maintenance mode', () => {
   test('Disabled (no maintenance planned)', async () => {
     mockMaintenanceKV({})
-    await handleRequest('https://de.serlo.org')
-    expectFetchToHaveBeenCalledWithRequest({
-      url: 'https://de.serlo.org',
-    } as Request)
+
+    await withMockedFetch([createApiResponse(), echoUrl], async () => {
+      const res = await handleUrl('https://de.serlo.org')
+      expect(await res.text()).toBe('https://de.serlo.org/')
```

Comment:
Why does the maintenance need to know anything about the API? Either Maintenance is in front of the proxy logic (i.e. when maintenance mode is on, we won't call the proxy) or behind (i.e. there is only a maintenance mode for non-frontend).

Reply:
And I don't like the refactor with mockedFetch. Previously, the code showed exactly what it did. Now we have some weird assertion on `res.text()` which is a fake response created via `echoUrl`...